package main

import "fmt"

func protegido() {
	defer func() {
		if r := recover(); r != nil {
			fmt.Println("ğŸ§¯ Se recuperÃ³ del pÃ¡nico:", r)
		}
	}()

	fmt.Println("Ejecutando funciÃ³n protegida")
	panic("ğŸ”¥ Error inesperado")
}

func main() {
	protegido()
	fmt.Println("âœ… El programa continÃºa despuÃ©s del recover")
}
